UPGRADING-HADOOP.md

```diff
@@ -0,0 +1,181 @@
+administrator to upgrade, downgrade and finalize a hadoop deployment. Currently
+rollback is not supported, and HDFS filesystem version upgrade is not supported (but
+will be implemented when a new filesystem version is released greater than the version
+which comes with 2.7.2)
+
+The actions and the commands they run are in line with the instructions and
+recommendations in this document:
+
+http://hadoop.apache.org/docs/r2.7.2/hadoop-project-dist/hadoop-HDFS/HDFSRollingUpgrade.html
+
+Therefore, when upgrading a namenode, datanode or resourcemanager, the following
+daemons will be stopped/started: 
+  namenode  
+  datanode
+  nodemanager
+  resourcemanager
+  jobhistoryserver
+
+While ALL hadoop binaries are upgraded by this process, neither journalnode nor
+zkfc (zookeeper) daemons are restarted by the upgrade process, as doing so can
+cause an HDFS outage. Further instructions for restarting these daemons after an
+upgrade are provided in the action definitions in: the datanode layer (for
+journalnode) and the namenode layer for zkfc. 
+
+  
+### Upgrade (HA)
+
+The apache documentation suggests the following process for performing a rolling
+upgrade - that is, an HDFS upgrade without downtime:
+
+  1. Prepare rolling upgrade
+  2. Upgrade namenodes
+  3. Upgrade (a subset of) datanodes
+  4. Finalize (or downgrade, or rollback*) 
+
+There is an additional 'safety' step with our hadoop implementations: you must
+config set the version that you would like to upgrade (or downgrade) to for each
+service you are going to upgrade. 
+
+Here are the general steps:
+
+1. Set hadoop_version config value for each service:
+
+
+    version=2.7.2
+    juju set namenode hadoop_version=$version
+    juju set slave hadoop_version=$version
+2. Prepare the rolling upgrade:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version prepare=true
+3. Upgrade both namenodes:
+
+
+    juju action do namenode/0 hadoop-upgrade version=$version
+    juju action do namenode/1 hadoop-upgrade version=$version 
+4. Upgrade datanodes:
+
+
+    juju action do slave/0 hadoop-upgrade version=$version
+    juju action do slave/1 hadoop-upgrade version=$version
+    juju action do slave/2 hadoop-upgrade version=$version
+
+If data persistence is important to you, you should upgrade a subset of your
+slaves/datanodes, perform some tests (e.g. check existing data) and then
+continue with the upgrade or rollback. 
+
+Once you have have finished upgrading your units (successfully or not) you will
+need to do one of the following:
+
+1. finalize - cleans up the HDFS filesystem, and marks the rolling upgrade as
+complete
+2. downgrade - restores a previous version of hadoop (after which you ***must
+still finalize***)
+3. rollback - **not yet available** will restore a previous version of hadoop
+and restore HDFS file system to the state it was in when 'prepare' was run. 
+
+
+### Upgrade (standalone)
```

Comment:
I am not sure what "standalone" means in this context. I cannot map it to  anything in "http://hadoop.apache.org/docs/r2.7.2/hadoop-project-dist/hadoop-hdfs/HdfsRollingUpgrade.html"?